x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/execution/search/QlSourceBuilder.java

```diff
@@ -66,14 +57,11 @@ public void addScriptField(String name, Script script) {
      */
     public void build(SearchSourceBuilder sourceBuilder) {
         sourceBuilder.trackScores(this.trackScores);
-        if (!sourceFields.isEmpty()) {
-            sourceBuilder.fetchSource(sourceFields.toArray(Strings.EMPTY_ARRAY), null);
-        }
-        docFields.forEach(field -> sourceBuilder.docValueField(field.field, field.format));
+        fetchFields.forEach(field -> sourceBuilder.fetchField(new FieldAndFormat(field.field, field.format, null)));
         scriptFields.forEach(sourceBuilder::scriptField);
     }
 
     public boolean noSource() {
```

Comment:
Likely this needs to go in a future PR since the notion of source is removed.

Reply:
Yes, I also believe there is room for improvement when it comes to removing code. There may be other places where things are not needed anymore.

Reply:
(seems to only be used once?)